source/common/upstream/outlier_detection_impl.cc

```diff
@@ -200,8 +206,47 @@ void DetectorImpl::onConsecutive5xxWorker(HostPtr host) {
 
 void DetectorImpl::onIntervalTimer() {
   SystemTime now = time_source_.currentSystemTime();
+  std::unordered_map<HostPtr, double> valid_sr_hosts;
+  std::vector<double> sr_data;
+  double sr_sum = 0;
+
   for (auto host : host_sinks_) {
     checkHostForUneject(host.first, host.second, now);
+
+    // Success Rate Outlier Detection
+    // First swap out the current bucket been written to, to keep data valid
+    host.second->updateCurrentSRBucket();
+
+    // If there are not enough hosts to begin with, don't do the work.
+    if (host_sinks_.size() >=
+        runtime_.snapshot().getInteger("outlier_detection.significant_host_threshold", 5)) {
+      Optional<double> host_sr = host.second->srAccumulator().getSR(
+          runtime_.snapshot().getInteger("outlier_detection.rq_volume_threshold", 100));
+      if (host_sr.valid()) {
+        valid_sr_hosts[host.first] = host_sr.value();
+        sr_data.emplace_back(host_sr.value());
+        sr_sum += host_sr.value();
+      }
+    }
+  }
+
+  if (valid_sr_hosts.size() >=
+      runtime_.snapshot().getInteger("outlier_detection.significant_host_threshold", 5)) {
+
+    // Calculate the statistics (mean, stdev). We are using mean to detect outliers.
+    double mean = sr_sum / sr_data.size();
+    double stdev = 0;
+    std::for_each(sr_data.begin(), sr_data.end(),
+                  [&stdev, mean](double& v) { stdev += std::pow(v - mean, 2); });
+    stdev /= sr_data.size();
+    stdev = std::sqrt(stdev);
+
+    for (auto host : valid_sr_hosts) {
+      if (host.second < mean - (2 * stdev)) {
```

Comment:
good to know. I would make the 2 a static const member var that way the name could indicate its use.